### PR TITLE
feat(tianmu): support the syntax of 'select..where like NULL' (#763)

### LINKED ISSUE #763: bug: Tianmu error ERROR 6 (HY000)，when execute select..where like NULL
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/stoneatom/stonedb/blob/stonedb-5.7-dev/CONTRIBUTING.md).

### Please confirm if bug report does NOT exists already ?

- [X] I confirm there is no existing issue for this

### Describe the problem

```shell
mysql> create table tlike (val varchar(255));
Query OK, 0 rows affected (0.00 sec)

mysql> insert into tlike values ('abcde');
Query OK, 1 row affected (0.00 sec)

mysql> select * from tlike where val like NULL;
ERROR 6 (HY000): The query includes syntax that is not supported by the storagngine. Either restructure the query with supported syntax, or enable the MySQLre::Query Path in config file to execute the query with reduced performance.
```

### Expected behavior

```shell
mysql> create table tlike (val varchar(255));
Query OK, 0 rows affected (0.03 sec)

mysql> insert into tlike values ('abcde');
Query OK, 1 row affected (0.01 sec)

mysql> select * from tlike where val like NULL;
Empty set (0.00 sec)
```

### How To Reproduce

_No response_

### Environment

ubuntu@ubuntu:~$ cat /etc/os-release
NAME="Ubuntu"
VERSION="20.04.4 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04.4 LTS"
VERSION_ID="20.04"

ubuntu@ubuntu:~$ /stonedb57/install/bin/mysqld --version
/stonedb57/install/bin/mysqld Ver 5.7.36-StoneDB for Linux on x86_64 (build-)
build information as follow:
Repository address: https://github.com/stoneatom/stonedb.git:stonedb-5.7-dev
Branch name: stonedb-5.7-dev
Last commit ID: https://github.com/stoneatom/stonedb/commit/5d996f9355b5241eaa54f3e4b652a60c9c229b0e
Last commit time: Date: Sat Oct 8 14:34:42 2022 +0800
Build time: Date: Fri 14 Oct 2022 12:28:00 PM UTC

### Are you interested in submitting a PR to solve the problem?

- [ ] Yes, I will!

### PR DESCRIPTION
**Pull request by isredstar**
## Summary about this PR

Issue Number: close #763 
add sql syntax：Arguments of like is NULL
add mtr for `like NULL`

## Tests Check List

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

## Changelog

- [x] New Feature
- [ ] Bug Fix
- [ ] Improvement
- [ ] Performance Improvement
- [ ] Build/Testing/CI/CD
- [ ] Documentation
- [ ] Not for changelog (changelog entry is not required)

## Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features

### CODE REVIEW

**Review by lujiashun — APPROVED** [review]
LGTM

File: mysql-test/suite/tianmu/t/issue763.test [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+create table tlike1 (id int, val varchar(255));
+INSERT INTO tlike1 VALUES(1,'a'), (1,'b'), (1,'c'), (2,'dd'), (3,'eee'), (4, NULL), (5, NULL);
+SELECT * FROM tlike1;
+SELECT * FROM tlike1 WHERE val is NULL;
+SELECT * FROM tlike1 WHERE val LIKE NULL;
+SELECT COUNT(*) FROM tlike1 WHERE val LIKE NULL;
+SELECT id, val FROM tlike1 WHERE id=1 and val LIKE NULL;
+SELECT id, val FROM tlike1 WHERE id=1 or val LIKE NULL;
+SELECT id, val FROM tlike1 WHERE id in (4, 5) and val LIKE NULL;
+SELECT id, val FROM tlike1 WHERE id in (4, 5) or val LIKE NULL;
```
Comment (RingsC) [inline mysql-test/suite/tianmu/t/issue763.test:23]: Would you like to add some cases for negtive condition, such as `not like`.

**Review by RingsC — CHANGES_REQUESTED** [review]

File: mysql-test/suite/tianmu/t/issue763.test (hunk 1 above)
Comment (isredstar) [reply mysql-test/suite/tianmu/t/issue763.test:23]: I'll make the requested changes!

**Review by RingsC — APPROVED** [review]
LGTM, Great Job.

**Pull request merged**